components/scripts/lib/gradle.sh

```diff
@@ -12,35 +12,45 @@ invoke_gradle() {
     cd "${project_dir}" > /dev/null 2>&1 || die "ERROR: Subdirectory ${project_dir} (set with --project-dir) does not exist in ${project_name}" "${INVALID_INPUT}"
   fi
 
-  args+=(
-    --init-script "${INIT_SCRIPTS_DIR}/develocity-injection.gradle"
-    --init-script "${INIT_SCRIPTS_DIR}/configure-build-validation.gradle"
-    -Ddevelocity.injection.init-script-name=develocity-injection.gradle
-    -Ddevelocity.injection-enabled=true
+  envs+=(
+    DEVELOCITY_INJECTION_INIT_SCRIPT_NAME=develocity-injection.gradle
+    DEVELOCITY_INJECTION_ENABLED=true
   )
 
   if [ "$enable_ge" == "on" ]; then
-    args+=(
-      -Dgradle.plugin-repository.url=https://plugins.gradle.org/m2
-      -Ddevelocity.plugin.version="3.14.1"
-      -Ddevelocity.ccud.plugin.version="2.0.2"
+    envs+=(
+      GRADLE_PLUGIN_REPOSITORY_URL=https://plugins.gradle.org/m2
+      DEVELOCITY_PLUGIN_VERSION="3.14.1"
+      DEVELOCITY_CCUD_PLUGIN_VERSION="2.0.2"
     )
   fi
 
   if [ -n "${ge_server}" ]; then
-    args+=(
-      -Ddevelocity.build-validation.url="${ge_server}"
-      -Ddevelocity.build-validation.allow-untrusted-server=false
+    envs+=(
+      DEVELOCITY_BUILD_VALIDATION_URL="${ge_server}"
+      DEVELOCITY_BUILD_VALIDATION_ALLOW_UNTRUSTED_SERVER=false
+    )
+  fi
+
+  if [ -n "${remote_build_cache_url}" ]; then
+    envs+=(
+      DEVELOCITY_BUILD_VALIDATION_REMOTEBUILDCACHEURL="${remote_build_cache_url}"
     )
   fi
```

Comment:
This `if` block is new. Can you explain?

Reply:
I now understand.

Reply:
👍 Jim brought it up too and it was discussed here, in case you missed it. 

https://github.com/gradle/gradle-enterprise-build-validation-scripts/pull/675#discussion_r1829896368